### ISSUE TITLE
SEP Debugger hangs for some programs for no apparent reason

### ISSUE THREAD

**Issue by heymchri** [codefori/vscode-ibmi#2296]
When I set a SEP in VS Code and then call the program from a green screen, all works OK - VS Code debug kicks in, shows the source and I can step through the program. So that confirms my setup is working.
 
However, when I set a SEP on one of our application exit programs, when that program gets executed as part of the web application workflow, the client hangs (likely waiting on the debugger to kick in), but the debugger just sits there, not bringing up the source etc. So somehow that communication is not working properly. The only thing I can do is to end the Debug Service Job to get the client to unlock.
 
Is there anywhere I can look for logs that might explain what's going on?
 
BTW - using plain old green screen to SEP debug the same works just fine.

I can provide a short video showing the issue if that helps.

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.13.5|
|Visual Studio Code version|1.94.2|
|Operating System|darwin_arm64|

<details>
<summary>Active extensions</summary>

```
.NET Install Tool (vscode-dotnet-runtime): 2.2.0
COBOL (cobol): 24.10.15
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.0
Dev Containers (remote-containers): 0.388.0
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.20.0
Excel Viewer (gc-excelviewer): 4.2.62
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
IBM i Debug (ibmidebug): 2.0.1
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Overtype (overtype): 0.5.0
Prettier - Code formatter (prettier-vscode): 11.0.0
Print (vscode-print): 0.13.2
Server Ready Action (debug-server-ready): 1.0.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
vscode-icons (vscode-icons): 12.9.0
```
</details>

<hr />

**_Not connected_** 🔌

**Comment by mkwan01** [comment]
You can enable the "Debug trace" setting and collect more log information from the following host files:

1. /QIBM/UserData/IBMiDebugService/DebugService_log.txt
2. /QIBM/UserData/IBMiDebugService/startDebugService_workspace/.metadata/.log

If you also have RDi, you can try the same debug scenario in RDi and tell us whether RDi can debug the program correctly. RDi and VSCode debug share some common components. If both of them have a problem, then the problem could be in the share component.

**Comment by heymchri** [comment]
Hi @mkwan01 I just installed an RDi evaluation copy to test - SEP debug works fine from there. 
Here's my scenario:
- CLLE program
- I set SEP in VS Code
- from green screen, I issue the command that has the CLLE as its CPP - debug editor shows up OK in VS Code
- in VS Code, I have an action that issues that same command - if I run the action, command hangs, debug editor doesn't show. Only way to get out of this is to end the Debug Service job, or to wait several minutes until something appears to time out and the command finally completes.
- in RDi, I set the SEP. I run the action in VS Code that issues the command - the debug editor in RDi shows up OK.

So I only have the issue in VS Code.

I have turned on Debug trace, and attached both host files (I renamed the .log file to log.txt).

[DebugService_log.txt](https://github.com/user-attachments/files/17821634/DebugService_log.txt)
[log.txt](https://github.com/user-attachments/files/17821647/log.txt)

**Comment by mkwan01** [comment]
Here are the messages from DebugService_log.txt:
```
Created SEP connection for user: HEYMCHR
No SEP connection for job: QP0ZSPWP/QSECOFR/413357
```

It shows that you logged on as HEYMCHR, but the job for the SEP is owned by QSECOFR. We should have a fix for this scenario in the next debug service update (around December 2024). The update allows you to modify the owner of the job where the SEP would be hit.

**Comment by heymchri** [comment]
Hi @mkwan01 I'm not quite sure why that SEP job would be owned by QSECOFR - there's nothing that I'm doing that would explain this. Is this something I have control over?

**Comment by mkwan01** [comment]
@heymchri I do not know how the job "QP0ZSPWP/QSECOFR/413357" is started. You can look at the "Current user identity" attribute of this job. This can be a different user profile than QSECOFR. If the Current user identity is your own user profile, then you will be able to do SEP debug automatically when you get the coming debug service PTF update. If the Current user identity is still QSECOFR, the coming debug update has a new feature that allows you to modify the user profile for a SEP. You will be able to do SEP debug after changing the SEP user profile to QSECOFR.

**Comment by heymchri** [comment]
Hi @mkwan01 I'm not quite sure this explains the behavior I'm seeing. When I set the SEP, it evidently is for my own user profile, and the SEP debugging works just fine if I call the program from a green screen session. When I call the program through some other action within VS Code, the debug hangs. I'll see if I can come up with a simple test scenario that you can try on your end.

**Comment by mkwan01** [comment]
@heymchri I think this scenario should be fixed in the coming debug service PTF update. You can wait for the PTF update and try again. The PTF update is likely available in early December.

**Comment by heymchri** [comment]
@mkwan01 Ok I'll test again once the debug service PTF is available. Will there be any notification when it becomes available?

**Comment by mkwan01** [comment]
This problem was fixed in debug client v2.0.2. The new debug client requires a debug service update via the following host PTFs:
7.3 PTF SJ02996
7.4 PTF SJ03026
7.5 PTF SJ03030

Please note that you need to regenerate the SSL certificate after updating the debug service.

@worksofliam

**Comment by heymchri** [comment]
@mkwan01 I can confirm that this issue is now resolved. Many thanks!!😊
